tokio/src/runtime/task/harness.rs

```diff
@@ -329,6 +330,20 @@ where
             }
         }));
 
+        // We catch panics here in case invoking a hook panics.
+        //
+        // We call this in a separate block so that it runs after the task appears to have
+        // completed and will still run if the destructor panics.
+        let _ = panic::catch_unwind(panic::AssertUnwindSafe(|| {
+            if let Some(f) = self.trailer().hooks.task_terminate_callback.as_ref() {
+                f(&TaskMeta {
+                    #[cfg(tokio_unstable)]
+                    id: self.core().task_id,
+                    _phantom: Default::default(),
+                })
+            }
+        }));
```

Comment:
Please scope the `catch_unwind` so we only pay its cost when the callback is actually set.

Reply:
@Darksonn addressed

Reply:
ty for the catch